mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/quota/CassandraPerUserMaxQuotaManager.java

```diff
@@ -19,116 +19,87 @@
     }
 
     @Override
-    public void setDefaultMaxStorage(long defaultMaxStorage) throws MailboxException {
-        session.execute(setDefaultMaxStorageStatement.bind(defaultMaxStorage));
+    public void removeMaxMessage(QuotaRoot quotaRoot) {
+        dao.removeMaxMessage(quotaRoot);
     }
 
     @Override
-    public void setDefaultMaxMessage(long defaultMaxMessageCount) throws MailboxException {
-        session.execute(setDefaultMaxMessageStatement.bind(defaultMaxMessageCount));
+    public void removeMaxStorage(QuotaRoot quotaRoot) {
+        dao.removeMaxStorage(quotaRoot);
     }
 
     @Override
-    public long getDefaultMaxStorage() throws MailboxException {
-        ResultSet resultSet = session.execute(getDefaultMaxStatement.bind()
-            .setString(CassandraDefaultMaxQuota.TYPE, CassandraDefaultMaxQuota.STORAGE));
-        if (resultSet.isExhausted()) {
-            return Quota.UNLIMITED;
-        }
-        return resultSet.one().getLong(CassandraDefaultMaxQuota.VALUE);
+    public void setDefaultMaxStorage(QuotaSize defaultMaxStorage) {
+        dao.setDefaultMaxStorage(defaultMaxStorage);
     }
 
     @Override
-    public long getDefaultMaxMessage() throws MailboxException {
-        ResultSet resultSet = session.execute(getDefaultMaxStatement.bind()
-            .setString(CassandraDefaultMaxQuota.TYPE, CassandraDefaultMaxQuota.MESSAGE));
-        if (resultSet.isExhausted()) {
-            return Quota.UNLIMITED;
-        }
-        return resultSet.one().getLong(CassandraDefaultMaxQuota.VALUE);
+    public void removeDefaultMaxStorage() {
+        dao.removeDefaultMaxStorage();
     }
 
     @Override
-    public long getMaxStorage(QuotaRoot quotaRoot) throws MailboxException {
-        ResultSet resultSet = session.execute(getMaxStorageStatement.bind(quotaRoot.getValue()));
-        if (resultSet.isExhausted()) {
-            return getDefaultMaxStorage();
-        }
-        return resultSet.one().getLong(CassandraMaxQuota.STORAGE);
+    public void setDefaultMaxMessage(QuotaCount defaultMaxMessageCount) {
+        dao.setDefaultMaxMessage(defaultMaxMessageCount);
     }
 
     @Override
-    public long getMaxMessage(QuotaRoot quotaRoot) throws MailboxException {
-        ResultSet resultSet = session.execute(getMaxMessageStatement.bind(quotaRoot.getValue()));
-        if (resultSet.isExhausted()) {
-            return getDefaultMaxMessage();
-        }
-        return resultSet.one().getLong(CassandraMaxQuota.MESSAGE_COUNT);
+    public void removeDefaultMaxMessage() {
+        dao.removeDefaultMaxMessage();
+    }
+
+    @Override
+    public Optional<QuotaSize> getDefaultMaxStorage() {
+        return dao.getDefaultMaxStorage();
+    }
+
+    @Override
+    public Optional<QuotaCount> getDefaultMaxMessage() {
+        return dao.getDefaultMaxMessage();
+    }
+
+    @Override
+    public Optional<QuotaSize> getMaxStorage(QuotaRoot quotaRoot) {
+        return dao.getMaxStorage(quotaRoot)
```

Comment:
completable future could help here, and allow choices like pre-fetching default quota for the (likely?) case where the user relies on default quotas...

Reply:
CompletableFuture help only one thing : latency.
For now, I thing that we can leave with the current code and refactor it later.